iterator/git.go

```diff
@@ -176,12 +179,15 @@ func (obj *Git) Validate() error {
 		return err // not that url.Parse ever really errors :/
 	}
 
-	if obj.Hash != "" && !plumbing.IsHash(obj.Hash) { // IsHash checks if len is 40
+	if obj.Hash != "" && (!plumbing.IsHash(obj.Hash) || strings.Contains(obj.Hash, separator)) { // IsHash checks if len is 40
 		return fmt.Errorf("provided hash is invalid")
 	}
 	// TODO: can we validate ref somehow?
 
 	if obj.Rev != "" {
+		if strings.Contains(obj.Rev, separator) {
```

Comment:
Does this need to be indented inside the parent `if obj.Rev != "" {` ?

Remember the goal is to always reduce indentation because it improves readability and makes it easier to stay under 80.

Reply:
Fixed